internal/provisioner/plain/controllers/bundleinstance_controller.go

```diff
@@ -246,48 +267,52 @@ func (r *BundleInstanceReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	for _, obj := range desiredObjects {
 		uMap, err := runtime.DefaultUnstructuredConverter.ToUnstructured(obj)
 		if err != nil {
-			meta.SetStatusCondition(&bi.Status.Conditions, metav1.Condition{
-				Type:    rukpakv1alpha1.TypeInstalled,
-				Status:  metav1.ConditionFalse,
-				Reason:  rukpakv1alpha1.ReasonCreateDynamicWatchFailed,
-				Message: err.Error(),
-			})
+			u.UpdateStatus(
+				updater.EnsureCondition(metav1.Condition{
+					Type:    rukpakv1alpha1.TypeInstalled,
+					Status:  metav1.ConditionFalse,
+					Reason:  rukpakv1alpha1.ReasonCreateDynamicWatchFailed,
+					Message: err.Error(),
+				}))
 			return ctrl.Result{}, err
 		}
 
-		u := &unstructured.Unstructured{Object: uMap}
+		uu := &unstructured.Unstructured{Object: uMap}
```

Comment:
nit: `uu` is a little confusing here. Maybe we could rename to something a bit more verbose?

```suggestion
		unstructuredObj := &unstructured.Unstructured{Object: uMap}
```

Reply:
Sounds good